plugins/catalog-import/src/components/ImportComponentForm.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorApiRef, useApi } from '@backstage/core';
+import { BackstageTheme } from '@backstage/theme';
+import {
+  Button,
+  FormControl,
+  FormHelperText,
+  TextField,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { useMountedState } from 'react-use';
+import { ComponentIdValidators } from '../util/validate';
+import { useGithubRepos } from '../util/useGithubRepos';
+import { ConfigSpec } from './ImportComponentPage';
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  form: {
+    alignItems: 'flex-start',
+    display: 'flex',
+    flexFlow: 'column nowrap',
+  },
+  submit: {
+    marginTop: theme.spacing(1),
+  },
+}));
+
+type Props = {
+  nextStep: () => void;
+  saveConfig: (configFile: ConfigSpec) => void;
+};
+
+export const RegisterComponentForm = ({ nextStep, saveConfig }: Props) => {
+  const { register, handleSubmit, errors, formState } = useForm({
+    mode: 'onChange',
+  });
+  const classes = useStyles();
+  const hasErrors = !!errors.componentLocation;
+  const dirty = formState?.isDirty;
+
+  const isMounted = useMountedState();
+  const errorApi = useApi(errorApiRef);
+  const { generateEntityDefinitions } = useGithubRepos();
+
+  const onSubmit = async (formData: Record<string, string>) => {
+    const { componentLocation: target } = formData;
+    try {
+      if (!isMounted()) return;
+
+      const repo = target.split('/').slice(-2).join('/');
```

Comment:
Using git-url-parse might be safer to use here.

Reply:
Fixed